corehq/apps/userreports/management/commands/queue_indicators.py

```diff
@@ -7,10 +7,13 @@
 
 
 class Command(BaseCommand):
+    help = "Queue all of the docs in the async indicators table for a given datasource"
 
-    def handle(self, **options):
-        indicators_by_domain_doc_type = defaultdict(list)
-        for indicator in AsyncIndicator.objects.all():
-            indicators_by_domain_doc_type[(indicator.domain, indicator.doc_type)].append(indicator)
-        for k, indicators in indicators_by_domain_doc_type.items():
-            _queue_indicators(indicators)
+    def add_arguments(self, parser):
+        parser.add_argument('data_source_id', help='id of data source to queue')
+        parser.add_argument('--agg-queue', dest='use_agg_queue', action='store_true',
+                            help='Use the aggregation queue instead of the async UCR queue to avoid competing with normal async UCR processing')
+
+    def handle(self, data_source_id, use_agg_queue, **options):
+        indicators = AsyncIndicator.objects.filter(indicator_config_ids__contains=[data_source_id])
```

Comment:
hey should, we also apply filter of `date_queued=None` here?

Reply:
Along with that does it also make sense to update the created date to future. just to avoid a situation where MPR is almost caught up and these indicators ends up being queued in both the queues.

Reply:
I don't think so. To the first question, the purpose of this command is to ensure that all indicators of a certain type are queued. If you are running this command (because you are backfilling a data source), you want to queue every document. The harm in double queueing something is minimal, it doesn't change the data and its possible with the tiny number of documents being actively processed by the pillow, and adding that filter prevents you from ever being able to use this to retry docs that had previously failed. 

I think that answer mostly applies to the second question as well, although if we were really at a point where the main queues were processing these, we probably wouldnt use this command anyway since the normal queues would take care of it. Furthermore, I think that is out of scope for this command. This command queues tasks that are in the AsyncIndicator table. What they look like in that table is up to how you added them. If you want them to have a date in the future, you can do that when you fill the table, but this command doesn't modify anything in the table. It just queues things. And again, the downside is pretty small, its like a small number of documents and it doesnt change the data. A batch of async indicators takes 5 minutes for the queue to process, so at worst we are double processing for that long.

Reply:
cool. Thanks for explanation.